configs/eslint-config-compass/index.js

```diff
@@ -1,143 +1,62 @@
-const path = require('path');
-
-const jsConfigurations = ['eslint:recommended'];
+'use strict';
 
-const tsConfigurations = [
-  ...jsConfigurations,
-  'plugin:@typescript-eslint/recommended',
-  'plugin:@typescript-eslint/recommended-requiring-type-checking',
-];
+const path = require('path');
+const shared = require('@mongodb-js/eslint-config-devtools');
+const common = require('@mongodb-js/eslint-config-devtools/common');
 
 const tsRules = {
-  '@typescript-eslint/no-unused-vars': 'error',
-  '@typescript-eslint/no-unsafe-assignment': 'off',
-  '@typescript-eslint/no-unsafe-call': 'off',
-  '@typescript-eslint/no-unsafe-member-access': 'off',
-  '@typescript-eslint/no-unsafe-return': 'off',
-  '@typescript-eslint/consistent-type-imports': [
-    'error',
-    { prefer: 'type-imports' },
-  ],
+  ...common.tsRules,
   // Newly converted plugins use `any` quite a lot, we can't enable the rule,
   // but we can warn so we can eventually address this
   '@typescript-eslint/no-unsafe-argument': 'warn',
 };
 
-const reactConfigurations = [
-  'plugin:react/recommended',
-  'plugin:react-hooks/recommended',
-  'plugin:jsx-a11y/recommended',
-];
-
-const testConfigurations = ['plugin:mocha/recommended'];
-
-const testRules = {
-  'mocha/no-exclusive-tests': 'error',
-  'mocha/no-hooks-for-single-case': 'off',
-  'mocha/no-setup-in-describe': 'off',
-  '@typescript-eslint/no-explicit-any': 'off',
-  '@typescript-eslint/no-empty-function': 'off',
-  '@typescript-eslint/no-unsafe-argument': 'off',
-  '@typescript-eslint/restrict-template-expressions': 'off',
+const tsOverrides = {
+  ...common.tsOverrides,
 };
+tsOverrides.rules = { ...tsRules };
 
-const javascriptParserOptions = {
-  parser: '@babel/eslint-parser',
-  parserOptions: {
-    ecmaVersion: 'latest',
-    requireConfigFile: false,
-    babelOptions: {
-      presets: [
-        require.resolve('@babel/preset-env'),
-        require.resolve('@babel/preset-react'),
-      ],
-    },
-  },
+const tsxRules = {
+  ...common.tsxRules,
+  '@typescript-eslint/no-unsafe-argument': 'warn',
 };
 
-const typescriptParserOptions = {
-  parser: '@typescript-eslint/parser',
-  parserOptions: {
-    ecmaFeatures: {
-      jsx: true,
-    },
-  },
+const tsxOverrides = {
+  ...common.tsxOverrides,
 };
+tsxOverrides.rules = { ...tsxRules };
```

Comment:
Ditto (and correspondingly below):

```suggestion
  rules: { ...tsRules },
};
```